### DISCUSSION TITLE
[Feature request]:WSL installation error QT

### DISCUSSION THREAD

**Discussion by Yuri-HubL** [OpenTOPAS/OpenTOPAS#34] [Feature-Request] [answered]
### Feature request

![1718851660689](https://github.com/OpenTOPAS/OpenTOPAS/assets/72288469/62efaa1f-8b47-412b-bb24-a1a1d4885665)

The second problem is that I don't know that I follow the instruction at the OpenTOPAS operating guidelines, whether it is right to download the Qt ...

![1718850908231](https://github.com/OpenTOPAS/OpenTOPAS/assets/72288469/4dcd8660-cabe-4e89-8bea-b161a03bcc7f)
I want to inform the developer, when I input/home/yuri/Applications/TOPAS OpenTOPAS - install/bin/TOPAS, I can get the correct installation as shown in the figure.So the problem is probably configuration.

My OpenTOPAS was downloaded using the following commands:
    **### cd ~/Applications/TOPAS
    git clone https://github.com/OpenTOPAS/OpenTOPAS.git**

But I didn't find it what path      /home/yuri/Applications/TOPAS/OpenTOPAS - install/Frameworks
![1718851890487](https://github.com/OpenTOPAS/OpenTOPAS/assets/72288469/7a922151-7481-49d9-a11b-5eb796a8e838)

What should I do? Hope can get your advice.

I am looking forward to your reply.

your sincerely,
Yuri

### TOPAS category

Geometry

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JNaokiDKondo** [comment]
Hello Yuri,
Apologies. For any Linux/Debian installation, no Frameworks folder will be created. Those are only for MacOS systems. We will update the WSL installation guide shortly.

In the case of WSL (Linux/Debian), you only need to specify the `LD_LIBRARY_PATH=~/Applications/TOPAS/OpenTOPAS-install/lib` folder and to copy the contents from the `GEANT4/geant4-install/lib` into this same folder.  See Step 8 of https://opentopas.readthedocs.io/en/latest/getting-started/WSL.html
The QT paths are defined during the Geant4 installation so they should come included when copying the libraries over. 

Another thing that might be wrong in your case is the environment variables (Step 9). And make sure to verify the contents of your `topas` shellScript as highlighted in the note at the end of the guide.

Greetings,
Naoki.

**Reply by Yuri-HubL** [reply]
Hi Naki,thank you for your reply.

I have some questions about your answer.

First, I think that when you say "**copy the contents from the geant4 /geant4-install/lib**", you are replacing the following code based on the original guide.
original:
    cd ~/Applications/TOPAS/OpenTOPAS-install/lib
    cp ~/Applications/GEANT4/geant4-install/lib/libG4* ./
change:
    cd ~/Applications/TOPAS/OpenTOPAS-install/lib
    cp ~/Applications/GEANT4/geant4-install/lib
 

Second, I'm not sure how I should modify the instructions:
    echo ’export QT_QPA_PLATFORM_PLUGIN_PATH=~/Applications/TOPAS/OpenTOPAS-install/Frameworks’ >> topas
At the moment, I'll leave he modified to the following manner
    echo ’export QT_QPA_PLATFORM_PLUGIN_PATH=~/Applications/TOPAS/OpenTOPAS-install’ >> topas
is this correct?
![(SM3TZU Z~GML173 BW}%KX](https://github.com/OpenTOPAS/OpenTOPAS/assets/72288469/13fc1698-8bf6-4b0c-8dac-fe795ad8a314)
I think my environment variables are consistent with the guidelines.

I am looking forward your reply.
Best,
Yuri

**Comment by JNaokiDKondo** [comment, marked as answer]
Hi Yuor,
I shortened the path in my answer, but in the WSL Quickstart guide is fully explained. Just go over to https://opentopas.readthedocs.io/en/latest/getting-started/WSL.html and follow the instructions.
No, you don't need the `QT_QPA_PLATFORM_PLUGIN_PATH` variable. Even if you have it, as long as you follow the WSL installation guide you should be able to run TOPAS with visualization.

Cheers,
Naoki.

**Reply by Yuri-HubL** [reply]
Hi Naoki,thank you for your reply.Now I can run it successfully.

![image](https://github.com/OpenTOPAS/OpenTOPAS/assets/72288469/77f3deaa-5a5c-497a-8944-9f344a72ce53)

![image](https://github.com/OpenTOPAS/OpenTOPAS/assets/72288469/f01d3d12-f2ad-4a0b-83f7-9dab3c4599cd)
To sum up, only need to **delete the QT_QPA_PLATFORM_PLUGIN_PATH** can run correctly.

I would also like to remind you that the quotation marks in the guide are wrong.

Cheers，
Yuri
